### PR TITLE
Remove blobstore from manifest builder

### PR DESCRIPTION
**Pull request by DavidSpek**
Rebase from https://github.com/distribution/distribution/pull/2782

> This PR aims to close https://github.com/distribution/distribution/issues/2150 by removing blobstore from schema2 manifest builder.
>
> The builder now take configuration Descriptor as parameter instead of a blobstore and media type.

@milosgajdos friendly ping since you were the one to review the original PR.

### CODE REVIEW

**Comment by DavidSpek** [comment]
Friendly ping @thaJeztah @squizzi

File: testutil/manifests.go [hunk 1]
```diff
@@ -76,9 +76,16 @@ func MakeSchema1Manifest(digests []digest.Digest) (distribution.Manifest, error)
...
 	blobStore := repository.Blobs(ctx)
-	builder := schema2.NewManifestBuilder(blobStore, schema2.MediaTypeImageConfig, []byte{})
-	for _, d := range digests {
-		builder.AppendReference(distribution.Descriptor{Digest: d})
+
+	var configJSON []byte
+
+	d, err := blobStore.Put(ctx, schema2.MediaTypeImageConfig, configJSON)
+	if err != nil {
+		return nil, fmt.Errorf("unexpected error storing content in blobstore %v", err)
```
Comment (squizzi) [inline testutil/manifests.go:84]: nit:
```suggestion
		return nil, fmt.Errorf("unexpected error storing content in blobstore: %v", err)
```

**Review by squizzi — APPROVED** [review]
lgtm

**Review by milosgajdos — APPROVED** [review]
LGTM. Would be nice to address the nits, though.

**Pull request merged**